.ci/cico_openshift_shost.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+set -ex
+
+#Stop execution on any error
+trap "catchFinish" EXIT SIGINT
+
+# Catch_Finish is executed after finish script.
+catchFinish() {
+  result=$?
+
+  if [ "$result" != "0" ]; then
+    echo "[ERROR] Please check the openshift ci artifacts"
+    getOCCheClusterLogs
+    exit 1
+  fi
+
+  echo "[INFO] Job finished Successfully.Please check openshift ci artifacts"
+  getOCCheClusterLogs
+
+  exit $result
+}
+
+# Define global environments
+function init() {
+  export SCRIPT=$(readlink -f "$0")
+  export SCRIPT_DIR=$(dirname "$SCRIPT")
+  export RAM_MEMORY=8192
+  export NAMESPACE="che"
+  export PLATFORM="openshift"
+
+  # Set operator root directory
+  if [[ ${WORKSPACE} ]] && [[ -d ${WORKSPACE} ]]; then
+    export OPERATOR_REPO=${WORKSPACE};
+  else
+    export OPERATOR_REPO=$(dirname "$SCRIPT_DIR");
+  fi
+
+    # CHE_OPERATOR_IMAGE is exposed in openshift ci pod. This image is build in every job and used then to deploy Che
+    # More info about how images are builded in Openshift CI: https://github.com/openshift/ci-tools/blob/master/TEMPLATES.md#parameters-available-to-templates
+    export OPERATOR_IMAGE=${CHE_OPERATOR_IMAGE}
+}
+
+# Get Token from single host mode deployment
+function getSingleHostToken() {
+    export KEYCLOAK_HOSTNAME=$(oc get routes/che -n ${NAMESPACE} -o jsonpath='{.spec.host}')
+    export TOKEN_ENDPOINT="https://${KEYCLOAK_HOSTNAME}/auth/realms/che/protocol/openid-connect/token"
+    export CHE_ACCESS_TOKEN=$(curl --data "grant_type=password&client_id=che-public&username=admin&password=admin" -k ${TOKEN_ENDPOINT} | jq -r .access_token)
+}
+
+# Utility to wait for a workspace to be started after workspace:create.
+function waitSingleHostWorkspaceStart() {
+  set +e
+  export x=0
+  while [ $x -le 180 ]
+  do
+    getSingleHostToken
+
+    chectl workspace:list
+    workspaceList=$(chectl workspace:list --chenamespace=${NAMESPACE})
+    workspaceStatus=$(echo "$workspaceList" | grep RUNNING | awk '{ print $4} ')
+
+    if [ "${workspaceStatus:-NOT_RUNNING}" == "RUNNING" ]
+    then
+      echo "[INFO] Workspace started successfully"
+      break
+    fi
+    sleep 10
+    x=$(( x+1 ))
+  done
+
+  if [ $x -gt 180 ]
+  then
+    echo "[ERROR] Workspace didn't start after 3 minutes."
+    exit 1
+  fi
+}
+
+# Utility to get che events and pod logs from openshift cluster
+function getOCCheClusterLogs() {
```

Comment:
can we use `chectl server:logs` to grab all logs and events ?

Reply:
I'll investigate. I thing even `chectl server:logs`. All pod logs will be saved in openshift ci artifacts